src/client/client-wrapper.ts

```diff
@@ -135,4 +135,30 @@ export class ClientWrapper {
     });
   }
 
+  /**
+   * Retrieves a single CampaignMember record for a given email address, including the
+   * provided field.
+   *
+   * @param {String} email - Email address of the CampaignMember record to retrieve.
+   * @param {String} field - CampaignMember field to include on the returned record.
+   */
+  // tslint:disable-next-line:max-line-length
+  public async findCampaignMemberByEmail(email: string, fields: string[]): Promise<Record<string, any>> {
+    await this.clientReady;
+    return new Promise((resolve, reject) => {
+      try {
+        this.client.sobject('CampaignMember').findOne(
+          { Email: email },
+          fields, (err, record) => {
+            if (err) {
+              reject(err);
+              return;
+            }
+            resolve(record);
+          });
+      } catch (e) {
+        reject(e);
+      }
+    });
+  }
```

Comment:
- [ ] Could you also add tests for this method to `test/client/client-wrapper.ts`?

Reply:
Added the tests. I cant cover the 2 lines involving the resolve part for deleteLead and findCampaignMemberByEmailAndCampaignId for some reason, even though findLeadById is completely covered which is almost the same tests as findCampaignMember